packages/elements/src/internals/machines/sign-up/sign-up.actors.ts

```diff
@@ -65,60 +72,73 @@ export const authenticateWithSignUpRedirect = fromPromise<void, AuthenticateWith
+      return Promise.resolve(clerk.client.signUp);
+    }
 
-// ================= prepareEmailAddressVerification ================= //
+    return await clerk.client.signUp.prepareVerification(params);
+  },
+);
 
-export type PrepareEmailAddressVerificationInput = WithClerk<WithParams<PrepareEmailAddressVerificationParams>>;
+export type AttemptVerificationInput = WithClerk<WithParams<AttemptVerificationParams>>;
 
-export const prepareEmailAddressVerification = fromPromise<SignUpResource, PrepareEmailAddressVerificationInput>(
-  async ({ input: { clerk, params } }) => await clerk.client.signUp.prepareVerification(params),
+export const attemptVerification = fromPromise<SignUpResource, AttemptVerificationInput>(
+  async ({ input: { clerk, params } }) => clerk.client.signUp.attemptVerification(params),
 );
 
-// ================= prepareWeb3WalletVerification ================= //
+// ================= Start / Continue ================= //
 
-export type PrepareWeb3WalletVerificationInput = WithClerk;
+export const SignUpAdditionalKeys = [
+  'firstName',
+  'lastName',
+  'emailAddress',
+  'username',
+  'password',
+  'phoneNumber',
+] as const;
 
-export const prepareWeb3WalletVerification = fromPromise<SignUpResource, PrepareWeb3WalletVerificationInput>(
-  async ({ input: { clerk } }) => await clerk.client.signUp.prepareWeb3WalletVerification(),
-);
+export type SignUpAdditionalKeys = (typeof SignUpAdditionalKeys)[number];
 
-// ================= createSignIn ================= //
+const updateSignUpKeys = new Set<SignUpAdditionalKeys>(SignUpAdditionalKeys);
 
-export type CreateSignUpInput = WithClerk<{ fields: FormMachineContext['fields'] }>;
+function isSignUpParam<T extends SignUpAdditionalKeys>(key: string): key is T {
+  return updateSignUpKeys.has(key as T);
+}
 
-export const createSignUp = fromPromise<SignUpResource, CreateSignUpInput>(async ({ input: { clerk, fields } }) => {
-  const emailAddress = fields.get('emailAddress');
-  const phoneNumber = fields.get('phoneNumber');
-
-  const firstName = fields.get('firstName');
-  const lastName = fields.get('lastName');
-  const username = fields.get('username');
-  const password = fields.get('password');
-
-  // TODO: Determine which takes priority
-
-  return clerk.client.signUp.create({
-    firstName: firstName?.value as string,
-    lastName: lastName?.value as string,
-    emailAddress: emailAddress?.value as string,
-    username: username?.value as string,
-    password: password?.value as string,
-    phoneNumber: phoneNumber?.value as string,
-    // unsafeMetadata: {} as SignUpUnsafeMetadata,
-    ...(password?.value
-      ? {
-          password: password.value as string,
-        }
-      : {}),
+function fieldsToSignUpParams<T extends SignUpCreateParams | SignUpUpdateParams>(
+  fields: FormFields,
+): Pick<T, SignUpAdditionalKeys> {
+  const params: SignUpUpdateParams = {};
+
+  // TODO: Determine what takes priority
```

Comment:
Not sure I understand that comment (I know it was there before) - isn't it obsolete now?

Reply:
I believe that you're right. I'll remove in my next PR so-as not to deal with any odd rebasing.